### ISSUE TITLE
 The 'Paste' distribution was not found and is required by the application

### ISSUE THREAD

**Issue by jugmac00** [zopefoundation/Zope#734]
There seem to be a bunch of related issues, but all closed #651 #452 #664 

I followed the official installation document at 
https://zope.readthedocs.io/en/latest/INSTALL.html
https://zope.readthedocs.io/en/latest/operation.html

**Steps to reproduce**
```
python3.7 -m venv zope
cd zope/
bin/pip install -U pip
bin/pip install Zope==4.1.2 -c https://zopefoundation.github.io/Zope/releases/4.1/constraints.txt
bin/mkwsgiinstance -d .
bin/runwsgi -v etc/zope.ini
```

The last command fails
```
bin/runwsgi -v etc/zope.ini 
Traceback (most recent call last):
  File "bin/runwsgi", line 8, in <module>
    sys.exit(main())
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/Zope2/Startup/serve.py", line 248, in main
    return command.run()
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/Zope2/Startup/serve.py", line 189, in run
    global_conf=vars)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/Zope2/Startup/serve.py", line 217, in loadapp
    return loadapp(app_spec, name=name, relative_to=relative_to, **kw)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 253, in loadapp
    return loadobj(APP, uri, name=name, **kw)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 277, in loadobj
    global_conf=global_conf)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 302, in loadcontext
    global_conf=global_conf)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 326, in _loadconfig
    return loader.get_context(object_type, name, global_conf)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 455, in get_context
    global_additions=global_additions)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 567, in _pipeline_app_context
    for name in pipeline[:-1]]
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 567, in <listcomp>
    for name in pipeline[:-1]]
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 459, in get_context
    section)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 481, in _context_from_use
    object_type, name=use, global_conf=global_conf)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 412, in get_context
    global_conf=global_conf)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 302, in loadcontext
    global_conf=global_conf)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 334, in _loadegg
    return loader.get_context(object_type, name, global_conf)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 625, in get_context
    object_type, name=name)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 645, in find_egg_entry_point
    pkg_resources.require(self.spec)
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/jugmac00/Tests/Zope4.1.2.installation/zope/lib/python3.7/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'Paste' distribution was not found and is required by the application
```

**Comment by icemac** [comment]
Hi @jugmac00 You are using the constraints of 4.1 to install Zope 4.1.2. But using the correct version of the file does not help either, so I created #735.

**Comment by jugmac00** [comment]
Uh, I missed this - there was a ton of back and forth while trying to reproduce the problems a user posted in https://community.plone.org/t/zmi-history-tab/10917

So, I think there is no good reason to keep this issue open? Zope 4.1.2 with constraints of 4.1 does not make a whole lot of sense?

Let's close it?

**Comment by dataflake** [comment]
I think I found a decent workaround in #735

**Comment by jugmac00** [comment]
closed => see #735
